napari/components/console.py

```diff
@@ -0,0 +1,18 @@
+from IPython.core.interactiveshell import InteractiveShell
+
+
+CODE = '''from skimage import filters
+output = filters.gaussian(viewer.layers[0].data, sigma=10)
+viewer.add_image(output)
+'''
+
+
+def call_ipython_code(code=CODE):
+    # get current running instance or create new instance
+    shell = InteractiveShell.instance()
```

Comment:
just FYI, the type of `shell` here, will depends on wether you are in a kernel of terminal IPython.
You could use this – or hasattr – to know what to do.